staging/src/k8s.io/kms/internal/plugins/mock/plugin.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"k8s.io/klog/v2"
+	"k8s.io/kms/pkg/hierarchy"
+	"k8s.io/kms/pkg/service"
+)
+
+var (
+	listenAddr = flag.String("listen-addr", "unix:///tmp/kms.socket", "gRPC listen address")
+	timeout    = flag.Duration("timeout", 5*time.Second, "gRPC timeout")
+)
+
+type remoteService struct {
+	keyID string
+}
+
+func main() {
+	flag.Parse()
+
+	addr, err := parseEndpoint(*listenAddr)
```

Comment:
You can use `ParseEndpoint` from `k8s.io/kms/pkg/util` after https://github.com/kubernetes/kubernetes/pull/116070 is merged!